src/main/java/uk/gov/pay/directdebit/notifications/services/UserNotificationService.java

```diff
@@ -0,0 +1,104 @@
+package uk.gov.pay.directdebit.notifications.services;
+
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.base.Stopwatch;
+import org.slf4j.Logger;
+import uk.gov.pay.directdebit.app.config.DirectDebitConfig;
+import uk.gov.pay.directdebit.app.logger.PayLoggerFactory;
+import uk.gov.pay.directdebit.gatewayaccounts.model.GatewayAccount;
+import uk.gov.pay.directdebit.gatewayaccounts.services.GatewayAccountService;
+import uk.gov.pay.directdebit.notifications.config.NotifyClientFactory;
+import uk.gov.pay.directdebit.payers.model.Payer;
+import uk.gov.pay.directdebit.payments.model.Transaction;
+import uk.gov.service.notify.NotificationClient;
+import uk.gov.service.notify.NotificationClientException;
+import uk.gov.service.notify.SendEmailResponse;
+
+import javax.inject.Inject;
+import java.util.HashMap;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import static java.lang.Runtime.getRuntime;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+
+public class UserNotificationService {
+    private static final Logger LOGGER = PayLoggerFactory.getLogger(UserNotificationService.class);
+
+    private String mandateFailedTemplateId;
+    private boolean emailNotifyGloballyEnabled;
+    private ExecutorService executorService;
+    private final MetricRegistry metricRegistry;
+    private final NotificationClient notificationClient;
+    private final DirectDebitConfig directDebitConfig;
+    private final GatewayAccountService gatewayAccountService;
+
+    @Inject
+    public UserNotificationService(DirectDebitConfig directDebitConfig,
+                                   NotificationClient notificationClient,
+                                   MetricRegistry metricRegistry,
+                                   GatewayAccountService gatewayAccountService) {
+        readMandateFailedEmailConfig(directDebitConfig.getNotifyConfig());
+        if (emailNotifyGloballyEnabled) {
+            int numberOfThreads = directDebitConfig.getExecutorServiceConfig().getThreadsPerCpu() * getRuntime().availableProcessors();
```

Comment:
Something that can be extracted in the configuration mapping class

Reply:
It's a bit tricky because of the runtime stuff, will take a look in the next PR